client/src/components/Login.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { makeStyles, Button, TextField, Grid } from "@material-ui/core";
+import LoginSignupContainer from "./LoginSignupContainer";
+
+// TODO Figure out where to move useStyles to avoid duplicate code
+const useStyles = makeStyles((theme) => ({
+  grid: {
+    flexGrow: 1,
+  },
+  form: {
+    "& .MuiTextField-root": {
+      margin: theme.spacing(1),
+      width: "40ch",
+    },
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  button: {
+    marginTop: 30,
+    marginBottom: 60,
+    borderRadius: 20,
+    color: "#FFFFFF",
+    backgroundColor: "#43DDC1",
+    width: "15ch",
+    padding: 10,
+  },
+  textfield: {
+    "& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline": {
+      borderColor: "purple",
+    },
+    "& label.Mui-focused": {
+      color: "purple",
+    },
+  },
+  link: {
+    color: "purple",
+    textDecoration: "none",
+  },
+  h1: {
+    fontSize: "xx-large",
+    marginBottom: 50,
+  },
+}));
+
+const Login = () => {
+  const classes = useStyles();
+
+  const [inputs, setInputs] = useState({ email: "", password: "" });
+
+  const handleChange = (name) => (event) => {
+    setInputs({ ...inputs, [name]: event.target.value });
```

Comment:
instead of creating a function that returns a function, I usually get the name from the event like:
const { name, value } = event.target; 

Then it doesn't require creating so many functions to do handleChange (right now you have a function for each field)